File: 프로그래머스/2/12953. N개의 최소공배수/N개의 최소공배수.java
class Solution {
    public int solution(int[] arr) {
        int answer = 0;
        
        if(arr.length == 1){
            return arr[0];
        }else{
            for(int i=0;i<arr.length-1;i++){
                arr[i+1] = arr[i] * arr[i+1] / gcd(arr[i], arr[i+1]);
            }
        }
        
        return arr[arr.length-1];
    }
    
    public int gcd(int a, int b){
        if(b == 0){
            return a;
        }
        else{
            return gcd(b, a%b);
        }
    }
}

File: 프로그래머스/2/12953. N개의 최소공배수/N개의 최소공배수.cpp
#include <bits/stdc++.h>

using namespace std;

int solution(vector<int> arr) {
    int answer = 0;
    
    sort(arr.rbegin(), arr.rend());
    
    int num = arr[0];
    
    int i=1;
    while (1) {
        int ok = 1;
        for (auto & n : arr) {
            answer = num * i;
            if (answer % n != 0) {
                ok = 0;
                break;
            }
        }
        
        if (ok) break;
        i++;
    }
    
    return answer;
}